File: CarlaErik/UvA.SoftCon.Questionnaire/AST/Model/Expressions/IntegerLiteral.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace UvA.SoftCon.Questionnaire.AST.Model.Expressions
{
    /// <summary>
    /// Represents a static, immutable integer value.
    /// </summary>
    public class IntegerLiteral : Literal<int>
    {
        public override NodeType Type
        {
            get
            {
                return NodeType.IntegerLiteral;
            }
        }

        public IntegerLiteral(int value, TextPosition position)
            : base(value, position) { }

        public override void Accept(IASTVisitor visitor)
        {
            visitor.Visit(this);
        }
    }
}

File: JeffSebastian/playground/src/main/java/nl/uva/se/ast/expression/literal/IntegerLiteral.java
package nl.uva.se.ast.expression.literal;

import nl.uva.se.visitor.ExpressionVisitor;

public class IntegerLiteral extends AbstractLiteral<Integer> {

	public IntegerLiteral(int lineNumber, int offset, Integer value) {
		super(lineNumber, offset, value);
	}

	@Override
	public <T> T accept(ExpressionVisitor<T> visitor) {
		return visitor.visit(this);
	}

}
